File: projeto/back/mais-consulta/src/main/java/com/bandtec/mais/consulta/domain/Usuario.java
package com.bandtec.mais.consulta.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import org.hibernate.Hibernate;
import org.hibernate.validator.constraints.br.CPF;

import javax.persistence.*;
import javax.validation.constraints.Email;
import java.util.Objects;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Builder
@Entity
@Table(name = "usuario", indexes = {
        @Index(name = "idx_usuario_cpf", columnList = "cpf"),
        @Index(name = "idx_usuario_password", columnList = "password")
})
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    private Integer idUsuario;

//    @CPF
    @Column(nullable = false)
    private String cpf;
//    @Email
    private String email;

    @JsonIgnore
    @Column(nullable = false)
    private String password;

    private String role;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        Usuario usuario = (Usuario) o;

        return Objects.equals(idUsuario, usuario.idUsuario);
    }

    @Override
    public int hashCode() {
        return 1225039686;
    }
}

File: projeto/back/maisconsulta/src/main/kotlin/com/bandtec/maisconsulta/domain/Usuario.kt
package com.bandtec.maisconsulta.domain

import org.hibernate.Hibernate
import javax.persistence.*

@Entity
@Table(name = "usuario")
data class Usuario(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_usuario")
    var idUsuario: Long? = null,

    @Column(name = "login")
    val login: String? = null,

    @Column(name = "password")
    var password: String? = null

) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || Hibernate.getClass(this) != Hibernate.getClass(other)) return false
        other as Usuario

        return idUsuario != null && idUsuario == other.idUsuario
    }

    override fun hashCode(): Int = 0

    @Override
    override fun toString(): String {
        return this::class.simpleName + "(idUsuario = $idUsuario , login = $login , password = $password )"
    }
}




